src/LanguageServer/Protocol/Workspaces/ILspMiscellaneousFilesWorkspaceProvider.cs

```diff
@@ -10,12 +10,19 @@
 
 namespace Microsoft.CodeAnalysis.LanguageServer;
 
+/// <summary>
+/// Allows the LSP server to create miscellaneous files documents.
+/// </summary>
+/// <remarks>
+/// No methods will be called concurrently, since we are dispatching LSP requests one at a time in the core dispatching loop.
+/// This does mean methods should be reasonably fast since they will block the LSP server from processing other requests while they are running.
+/// </remarks>
 internal interface ILspMiscellaneousFilesWorkspaceProvider : ILspService
 {
     /// <summary>
-    /// Returns the actual workspace that the documents are added to or removed from.
+    /// Returns whether the document is one that came from this provider.
     /// </summary>
-    Workspace Workspace { get; }
+    ValueTask<bool> IsProvidedDocumentAsync(TextDocument document);
```

Comment:
`WorkspaceContains` might be a reasonable name for this?

Reply:
But it's not really depending on a workspace anymore; maybe the "IsDocumentManagedByThis" the right method name or something?